src/neural/cuda/winograd_helper.inc

```diff
@@ -419,6 +419,24 @@ __device__ __forceinline__ float warpReduce(float x) {
   return x;
 }
 
+// fast max reduction for the warp
+__device__ __forceinline__ float warpMax(float x) {
+#pragma unroll
+  for (int mask = 16; mask > 0; mask >>= 1)
+    x = max(x, __shfl_xor_sync(0xFFFFFFFF, x, mask));
+
+  return x;
+}
+
+// atomic max implementation for floats
+__device__ __forceinline__ float atomicMaxFloat (float * addr, float val) {
```

Comment:
Stray space after atomicMaxFloat.  Also float * addr should be float* addr
Casts below should be int* and unsigned int* similarly (no space between type name and the * character)